src/main/java/mezz/jei/load/registration/SubtypeRegistration.java

```diff
@@ -47,10 +71,22 @@ public boolean hasSubtypeInterpreter(ItemStack itemStack) {
 		return interpreters.containsKey(item);
 	}
 
+	@Override
+	public boolean hasSubtypeInterpreter(FluidStack fluidStack) {
+		ErrorUtil.checkNotEmpty(fluidStack);
```

Comment:
For ItemStack, we check if it's empty because empty ones contain "air". Do empty fluidstacks have similar behavior? If not, we probably just want to check that this is not null.

Reply:
There IS a Fluid version of "Air" (net.minecraft.fluid.EmptyFluid), which a FluidStack will return if isEmpty is true. That is, a FluidStack, like an ItemStack, shouldn't be null.